lib/email_cook.rb

```diff
@@ -25,17 +21,21 @@ def add_quote(result, buffer)
 
   def link_string!(line, unescaped_line)
     unescaped_line = unescaped_line.strip
-    unescaped_line.scan(EmailCook.url_regexp).each do |m|
-      url = m[0]
-
-      if unescaped_line == url
-        # this could be oneboxed
-        val = %|<a href="#{url}" class="onebox" target="_blank">#{url}</a>|
-      else
-        val = %|<a href="#{url}">#{url}</a>|
+    line.gsub!(/\S+/) do |str|
+      if str.match?(/^(https?:\/\/)[\S]+$/i)
```

Comment:
Does that support URLs which starts with `www.`?

Reply:
No, it doesn't. The old regex detected URLs starting with `www` or `www2`, but the rest of the code didn't handle that well and created invalid links. I think the new code is actually an improvement.

Reply:
Pretty hard? I can't think of a good way of ensuring that. :thinking:
I'm not even sure that this code is still being used. I couldn't find any place in our code where we are setting the `cook_method` to `email`. This might actually be dead code.